### PR TITLE
[WFLY-13806] Add subsystem parsing of persistence.xml documents in th…

### PR DESCRIPTION
**Pull request by bstansberry**
…e EE 9 namespace

https://issues.redhat.com/browse/WFLY-13806

### CODE REVIEW

**Review by scottmarlow — APPROVED** [review]

File: jpa/subsystem/src/test/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParserTestCase.java [hunk 1]
```diff
@@ -59,4 +64,57 @@ public void testVersion() throws Exception {
         String version = metadata.getPersistenceXMLSchemaVersion();
         assertEquals("1.0", version);
     }
+
+    @Test
```
Comment (scottmarlow) [inline jpa/subsystem/src/test/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParserTestCase.java:68]: Thanks for adding this cool test, great idea!

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java [hunk 2]
```diff
@@ -94,8 +94,10 @@ public static PersistenceUnitMetadataHolder parse(final XMLStreamReader reader,
                 version = Version.JPA_2_2;
             } else if ("2".equals(versionString)) {
                 version = Version.JPA_2_0;
+            } else if ("3.0".equals(versionString)) {
+                version = Version.JPA_3_0;
             } else {
-                version = Version.JPA_2_1;
+                version = Version.JPA_2_1; // TODO is this still the right default?
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java:100]: Good question, for EE 8 I would like to default to the equivalent Jakarta (EE 8) Persistence 2.2 (http://xmlns.jcp.org/xml/ns/persistence/persistence_2_2.xsd) as parsePU() is doing.

For EE 9, I would like to default to 3.0 as you did below for parsePU().

Comment (bstansberry) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java]: @scottmarlow I added a static initializer to see if an EE 9 api class is available and determine the default based on that.

As part of doing that I also centralized the lookup logic and the version # strings a bit.

Comment (bstansberry) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java]: @scottmarlow Does the change I describe ^^^ look good?

**Comment by darranl** [comment]
@scottmarlow Does your review still stand?  Looks like Brian added something post review to double check.

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java [hunk 3]
```diff
@@ -49,6 +49,19 @@
 public class PersistenceUnitXmlParser extends MetaDataElementParser {
     // cache the trace enabled flag
     private static final boolean traceEnabled = ROOT_LOGGER.isTraceEnabled();
+    private static final Version DEFAULT_VERSION;
+
+    static {
+        Version defaultVersion;
+        try {
+            // Try and load a jakarta namespace JPA API class to see if we're EE 8 or a later EE
+            PersistenceUnitXmlParser.class.getClassLoader().loadClass("jakarta.persistence.SharedCacheMode");
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java:58]: Cool idea to detect the default as Jakarta EE 8 Persistence 2.2 versus EE 9 Persistence 3.0!

**Review by scottmarlow — APPROVED** [review]
Thanks Brian!

**Pull request merged**